File: src/main/java/com/gpch/login/model/Place.java
package com.gpch.login.model;


import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;

@Data
@Entity
@Table(name = "place")
public class Place {
    @Id
    @Column(name = "Id")
    Integer id;
    @NotBlank(message = "Wpisz coś kurwa")
    @Column(name = "name")
    String name;
    @NotBlank
    @Column(name = "address")
    String address;
    @Column(name = "contact_number")
    Integer contactNumber;
    @Column(name = "email")
    String email;
    @Column(name = "website")
    String website;
    @Column(name = "verified")
    Boolean verified;
    @Enumerated(EnumType.STRING)
    @Column(name = "sports")
    Sport sports;


    enum Sport {
        FOOTBALL, VOLLEYBALL, BASKETBALL
    }
}

File: src/main/java/com/gpch/login/model/Place.kt
package com.gpch.login.model


import javax.persistence.*


@Entity
@Table(name = "place")
data class Place constructor(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        @Column(name = "Id")
        var id: Int? = null,
        @Column(name = "name")
        var name: String? = null,
        @Column(name = "street")
        var street: String? = null,
        @Column(name = "street_no")
        var streetNo: Int? = null,
        @Column(name = "city")
        var city: String? = null,
        @Column(name = "contact_number")
        var contactNumber: Int? = null,
        @Column(name = "email")
        var email: String? = null,
        @Column(name = "website")
        var website: String? = null,
        @Column(name = "verified")
        var verified: Boolean? = null,
        @ElementCollection(targetClass = Sport::class)
//        @JoinTable(name = "tblInterests", joinColumns = [JoinColumn(name = "personID")])
        @Column(name = "sports")
        @Enumerated(EnumType.STRING)
        var sports: MutableList<Sport>? = null

) {
    enum class Sport {
        FOOTBALL, VOLLEYBALL, BASKETBALL
    }

}
